tests/dryrun/install_test.go

```diff
@@ -0,0 +1,359 @@
+		}
+	}
+
+	// --- validate host preflight spec --- //
+	foundFSPCollector := false
+	foundLAMPortCollector := false
+	foundKotsadmPortCollector := false
+
+	for _, hc := range dr.HostPreflightSpec.Collectors {
+		if hc.FilesystemPerformance != nil {
+			foundFSPCollector = true
+			assert.Equal(t, "/var/lib/embedded-cluster/k0s/etcd", hc.FilesystemPerformance.Directory)
+		}
+		if hc.TCPPortStatus != nil {
+			if hc.TCPPortStatus.CollectorName == "Local Artifact Mirror Port" {
+				foundLAMPortCollector = true
+				assert.Equal(t, 50000, hc.TCPPortStatus.Port)
+			}
+			if hc.TCPPortStatus.CollectorName == "Kotsadm Node Port" {
+				foundKotsadmPortCollector = true
+				assert.Equal(t, 30000, hc.TCPPortStatus.Port)
+			}
+		}
+	}
+
+	assert.True(t, foundFSPCollector, "FilesystemPerformance collector not found")
+	assert.True(t, foundLAMPortCollector, "Local Artifact Mirror Port collector not found")
+	assert.True(t, foundKotsadmPortCollector, "Kotsadm Node Port collector not found")
+
+	// --- validate metrics --- //
+	foundInstallationStarted := false
+	foundInstallationSucceeded := false
+	for _, m := range dr.Metrics {
+		if m.Title == "InstallationStarted" {
+			foundInstallationStarted = true
+		}
+		if m.Title == "InstallationSucceeded" {
+			foundInstallationSucceeded = true
+		}
+	}
+	if !foundInstallationStarted {
+		t.Errorf("InstallationStarted metric not found")
+	}
+	if !foundInstallationSucceeded {
+		t.Errorf("InstallationSucceeded metric not found")
+	}
+
+	// --- validate cluster resources exist --- //
+	kcli, err := dr.KubeClient()
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
+
+	assertConfigMapExists(t, kcli, "embedded-cluster-host-support-bundle", "kotsadm")
+	assertSecretExists(t, kcli, "kotsadm-password", "kotsadm")
+	assertSecretExists(t, kcli, "cloud-credentials", "velero")
+
+	// --- validate installation object --- //
+	in, err := kubeutils.GetLatestInstallation(context.TODO(), kcli)
+	if err != nil {
+		t.Fatalf("failed to get latest installation: %v", err)
+	}
+
+	assert.Equal(t, "80-32767", in.Spec.Network.NodePortRange)
+	assert.Equal(t, "10.244.0.0/16", dr.Flags["cidr"])
+	assert.Equal(t, "10.244.0.0/17", in.Spec.Network.PodCIDR)
+	assert.Equal(t, "10.244.128.0/17", in.Spec.Network.ServiceCIDR)
+	assert.Equal(t, 30000, in.Spec.RuntimeConfig.AdminConsole.Port)
+	assert.Equal(t, "/var/lib/embedded-cluster", in.Spec.RuntimeConfig.DataDir)
+	assert.Equal(t, 50000, in.Spec.RuntimeConfig.LocalArtifactMirror.Port)
+	assert.Equal(t, "ec-install", in.ObjectMeta.Labels["replicated.com/disaster-recovery"])
+
+	// --- validate k0s cluster config --- //
+	stdout, err = exec.Command("cat", defaults.PathToK0sConfig()).Output()
+	if err != nil {
+		t.Fatalf("fail to get k0s config: %v", err)
+	}
+
+	k0sConfig := k0sv1beta1.ClusterConfig{}
+	if err := yaml.Unmarshal([]byte(stdout), &k0sConfig); err != nil {
+		t.Fatalf("fail to unmarshal k0s config: %v", err)
+	}
```

Comment:
you could probably make convenience functions in the dryrun package for this

Reply:
can you elaborate? this file gets written to the filesystem, it's not captured by dryrun